card/mod/core/set/all/fetch.rb

```diff
@@ -88,7 +88,7 @@ def fetch_name *mark
     if (card = quick_fetch(mark))
       card.name
     elsif block_given?
-      yield.name
+      yield
```

Comment:
This needs a `to_name` conversion so that you don't get the wrong type if the block returns a string (happens in wikirate). It's also wrong in the rescue block